error building with Apple clang 8.1.0

I currently can not build latest xtensor trunk with Apple clang 8.1.0 (AFAIK based on clang 3.9.1) with tests enabled. I get the following error:
```
FAILED: test/CMakeFiles/test_xtensor_lib.dir/test_xreducer.cpp.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DXTENSOR_ENABLE_ASSERT -DXTENSOR_ENABLE_CHECK_DIMENSION -DXTENSOR_USE_XSIMD -I/Users/bdaci01/builds/DaPSmLFR/0/xtensor/include -isystem /data/Debug/include -ggdb3 -fno-omit-frame-pointer -O1 -march=core2 -msse4.1 -mtune=sandybridge -mmacosx-version-min=10.12  -DDEBUG -std=c++14 -march=native -Wunused-parameter -Wextra -Wreorder -Wconversion -Wsign-conversion -Wold-style-cast -Wunused-variable -O3 -DNDEBUG -fPIE   -std=gnu++14 -MD -MT test/CMakeFiles/test_xtensor_lib.dir/test_xreducer.cpp.o -MF test/CMakeFiles/test_xtensor_lib.dir/test_xreducer.cpp.o.d -o test/CMakeFiles/test_xtensor_lib.dir/test_xreducer.cpp.o -c /Users/bdaci01/builds/DaPSmLFR/0/xtensor/test/test_xreducer.cpp
In file included from /Users/bdaci01/builds/DaPSmLFR/0/xtensor/test/test_xreducer.cpp:10:
In file included from /Users/bdaci01/builds/DaPSmLFR/0/xtensor/include/xtensor/xarray.hpp:19:
In file included from /Users/bdaci01/builds/DaPSmLFR/0/xtensor/include/xtensor/xcontainer.hpp:21:
In file included from /Users/bdaci01/builds/DaPSmLFR/0/xtensor/include/xtensor/xiterable.hpp:12:
/Users/bdaci01/builds/DaPSmLFR/0/xtensor/include/xtensor/xiterator.hpp:1062:38: error: invalid operands to binary expression ('const xt::const_array<unsigned long, 3>' and 'const xt::const_array<unsigned long, 3>')
        XTENSOR_ASSERT(this->shape() == rhs.shape());
                       ~~~~~~~~~~~~~ ^  ~~~~~~~~~~~
```
Does this mean anything to you? Is my compiler too old? Anything I can do to help debug this issue?

--- a/include/xtensor/xstorage.hpp
+++ b/include/xtensor/xstorage.hpp
@@ -1515,7 +1515,44 @@ namespace xt
 #undef GCC4_FALLBACK
 
 
-// Workaround for rebind_container problems on GCC 8  with C++17 enabled
+    template <class T, std::size_t N>
+    inline bool operator==(const const_array<T, N>& lhs, const const_array<T, N>& rhs)
+    {
+        return std::equal(lhs.cbegin(), lhs.cend(), rhs.cbegin());
+    }
+
+    template <class T, std::size_t N>
+    inline bool operator!=(const const_array<T, N>& lhs, const const_array<T, N>& rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    template <class T, std::size_t N>
+    inline bool operator<(const const_array<T, N>& lhs, const const_array<T, N>& rhs)
+    {
+        return std::lexicographical_compare(lhs.begin(), lhs.end(),
+                                            rhs.begin(), rhs.end());
+    }
+
+    template <class T, std::size_t N>
+    inline bool operator<=(const const_array<T, N>& lhs, const const_array<T, N>& rhs)
+    {
+        return !(lhs > rhs);
+    }
+
+    template <class T, std::size_t N>
+    inline bool operator>(const const_array<T, N>& lhs, const const_array<T, N>& rhs)
+    {
+        return rhs < lhs;
+    }
+
+    template <class T, std::size_t N>
+    inline bool operator>=(const const_array<T, N>& lhs, const const_array<T, N>& rhs)
+    {
+        return !(lhs < rhs);
+    }
+
+// Workaround for rebind_container problems on GCC 8 with C++17 enabled
 #if defined(__GNUC__) && __GNUC__ > 6 && !defined(__clang__) && __cplusplus >= 201703L
     template <class X, class T, std::size_t N>
     struct rebind_container<X, aligned_array<T, N>>

--- a/test/test_xreducer.cpp
+++ b/test/test_xreducer.cpp
@@ -631,7 +631,6 @@ namespace xt
         auto res0 = xt::sum(x, { 0 }, xt::keep_dims);
         EXPECT_EQ(res0.shape()[0], size_t(1));
         EXPECT_EQ(res0.shape()[1], size_t(1));
-        int zzz = res0(0, 0);
         EXPECT_EQ(res0(0, 0), 0);
 
         auto res1 = xt::sum(x, { 1 }, xt::keep_dims);
@@ -640,9 +639,8 @@ namespace xt
         EXPECT_EQ(res1.size(), size_t(0));
 
         auto res2 = xt::sum(x, xt::keep_dims);
-        EXPECT_EQ(res2.shape()[0], 1);
-        EXPECT_EQ(res2.shape()[1], 1);
-        int xxx = res2(0, 0);
+        EXPECT_EQ(res2.shape()[0], size_t(1));
+        EXPECT_EQ(res2.shape()[1], size_t(1));
         EXPECT_EQ(res2(0, 0), 0);
     }
 }

